folium/plugins/semicircle.py

```diff
@@ -0,0 +1,108 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import (absolute_import, division, print_function)
+
+import json
+
+from branca.element import Figure, JavascriptLink
+
+from folium.map import Marker
+from folium.utilities import _validate_location
+
+from jinja2 import Template
+
+
+class SemiCircle(Marker):
+    """
+    Creates a Semicircle plugin to append into a map with
+    Map.add_plugin.
+    Use (direction and arc) or (start_angle and stop_angle)
+    Parameters
+    ----------
+    location: tuple of length 2, default None
+        The latitude and longitude of the marker.
+        If None, then the middle of the map is used.
+    radius: int, default 0
+        Radius of semicircle in meters
+    direction: int, default 0
+        Heading of direction angle value between 0 and 360 degrees
+    arc: int, default 0
+        Heading of arc angle value between 0 and 360 degrees.
+    start_angle: int, default 0
+        Heading of the start angle value between 0 and 360 degrees
+    stop_angle: int, default 0
+        Heading of the stop angle value between 0 and 360 degrees.
+    fill_color: string, default '#3388ff' (blue)
+        Fill color
+    fill_opacity: float 0-1, default 0.5
+        Fill opacity
+    color: string, default '#3388ff' (blue)
+        Line color
+    opacity: float 0-1, default 1
+        Line opacity
+    **kwargs:
+        Passed to Marker so a popup and tooltip can be provided.
+
+    """
+    _template = Template(u"""
+            {% macro script(this, kwargs) %}
+                if ({{this.direction}} || {{this.arc}}) {
```

Comment:
Question: can you make this code more [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)? If not that's okay, but if you want to do that that would be welcome.

Reply:
I see what you mean. Would you do this in python, by breaking the string into parts and using one part twice or would you do it in "JavaScript" within the template. JavaScript is not my forte to say the least. I tried a couple of things but they did not work. I realised today that maybe you were intending to do it directly with python. What do you suggest?

Reply:
I checked and what I think is the most sane approach is this:

- Pass `start_angle` and `stop_angle` into `parse_options`. If they are not None they will be added to the options.
- Store `direction` and `arc` in an optional tuple in the instance: `self.direction = (direction, arc) if direction and arc else None`
- In the template check if `this.direction` is set and apply `setDirection` if so: `{% if this.direction %}.setDirection({{ this.direction | tojson }}){% endif %}`

This way we don't need the big if/else.

In the Python part you already have a check if both angles and direction are not set at the same time.

Reply:
I like that, it's elegant and reduces the template considerably.